agents/ten_packages/extension/openai_v2v_python/client.py

```diff
@@ -0,0 +1,175 @@
+import asyncio
+import base64
+import json
+import os
+from typing import Any, AsyncGenerator
+
+import uuid
+import aiohttp
+from . import messages
+
+from .log import logger
+
+DEFAULT_MODEL = "gpt-4o-realtime-preview"
+
+DEFAULT_INSTRUCTION = '''
+You are a helpful voice assistant, user can see transcription through the chat box. Your name is TEN Agent. You are built by TEN Framework which is a powerful realtime multimodal agent framework. User's input will mainly be {language}, and your response must be {language}.
+{tools}
+'''
+
+def smart_str(s: str, max_field_len: int = 128) -> str:
+    """parse string as json, truncate data field to 128 characters, reserialize"""
+    try:
+        data = json.loads(s)
+        if "delta" in data:
+            key = "delta"
+        elif "audio" in data:
+            key = "audio"
+        else:
+            return s
+
+        if len(data[key]) > max_field_len:
+            data[key] = data[key][:max_field_len] + "..."
+        return json.dumps(data)
+    except json.JSONDecodeError:
+        return s
+
+
+def generate_client_event_id() -> str:
+    return str(uuid.uuid4())
+
+class RealtimeApiConfig:
+    def __init__(
+            self,
+            base_uri: str = "wss://api.openai.com",
+            api_key: str | None = None,
+            path: str = "/v1/realtime",
+            verbose: bool = False,
+            model: str=DEFAULT_MODEL,
+            language: str = "en-US",
+            system_message: str=DEFAULT_INSTRUCTION,
+            temperature: float =0.5,
+            max_tokens: int =1024,
+            voice: messages.Voices = messages.Voices.Alloy,
+            server_vad:bool=True,
+        ):
+        self.base_uri = base_uri
+        self.api_key = api_key
+        self.path = path
+        self.verbose = verbose
+        self.model = model
+        self.language = language
+        self.system_message = system_message
+        self.temperature = temperature
+        self.max_tokens = max_tokens
+        self.voice = voice
+        self.server_vad = server_vad
+    
+    def build_ctx(self) -> dict:
+        return {
+            "language": self.language
+        }
+
+class RealtimeApiClient:
```

Comment:
It seems somehow duplicated with the `realtime/connection.py`?

Reply:
removed